agent/consul/fsm/commands_oss_test.go

```diff
@@ -1347,6 +1347,111 @@ func TestFSM_ConfigEntry(t *testing.T) {
 	}
 }
 
+func TestFSM_ConfigEntry_StatusCAS(t *testing.T) {
+	t.Parallel()
+
+	logger := testutil.Logger(t)
+	fsm, err := New(nil, logger)
+	require.NoError(t, err)
+
+	// Create a simple config entry
+	entry := &structs.APIGatewayConfigEntry{
+		Kind:           structs.APIGateway,
+		Name:           "global",
+		EnterpriseMeta: *structs.DefaultEnterpriseMetaInDefaultPartition(),
+		Status: structs.Status{
+			Conditions: []structs.Condition{{
+				Status: "Foo",
+			}},
+		}}
+
+	// Create a new request.
+	req := &structs.ConfigEntryRequest{
+		Op:    structs.ConfigEntryUpsertCAS,
+		Entry: entry,
+	}
+
+	{
+		buf, err := structs.Encode(structs.ConfigEntryRequestType, req)
+		require.NoError(t, err)
+		resp := fsm.Apply(makeLog(buf))
+		if _, ok := resp.(error); ok {
+			t.Fatalf("bad: %v", resp)
+		}
+	}
+
+	// Verify it's in the state store.
+	{
+		_, config, err := fsm.state.ConfigEntry(nil, structs.APIGateway, "global", nil)
+		require.NoError(t, err)
+		entry.RaftIndex.CreateIndex = 1
+		entry.RaftIndex.ModifyIndex = 1
+		require.Equal(t, entry.DefaultStatus(), config.(*structs.APIGatewayConfigEntry).GetStatus())
+	}
+
+	// do a status update
+	entry.Status = structs.Status{
+		Conditions: []structs.Condition{{
+			Status: "Foo",
+		}},
+	}
+	req = &structs.ConfigEntryRequest{
+		Op:    structs.ConfigEntryUpsertStatusCAS,
+		Entry: entry,
+	}
+
+	{
+		buf, err := structs.Encode(structs.ConfigEntryRequestType, req)
+		require.NoError(t, err)
+		resp := fsm.Apply(makeLog(buf))
+		if _, ok := resp.(error); ok {
+			t.Fatalf("bad: %v", resp)
+		}
+	}
+
+	// Verify it's in the state store.
+	{
+		_, config, err := fsm.state.ConfigEntry(nil, structs.APIGateway, "global", nil)
+		require.NoError(t, err)
+		entry.RaftIndex.CreateIndex = 2
+		entry.RaftIndex.ModifyIndex = 2
+		conditions := config.(*structs.APIGatewayConfigEntry).Status.Conditions
+		require.Len(t, conditions, 1)
+		require.Equal(t, "Foo", conditions[0].Status)
+	}
+
+	// do a regular update and make sure the status gets reset
```

Comment:
I think this is the intention of this test, makes it a little clearer to me

```suggestion
	// attempt to change the status w/ a regular update and make sure it's ignored
```